jenkins/spark-premerge-build.sh

```diff
@@ -160,7 +160,17 @@ ci_2() {
     # Test a portion of cases for non-UTC time zone because of limited GPU resources.
     # Here testing: parquet scan, orc scan, csv scan, cast, TimeZoneAwareExpression, FromUTCTimestamp
     # Nightly CIs will cover all the cases.
-    TZ=Iran TEST='test_parquet_read_round_trip or test_read_round_trip or test_basic_csv_read or test_cast_string_ts_valid_format or test_unix_timestamp or test_from_utc_timestamp' ./integration_tests/run_pyspark_from_build.sh
+    #
+    # potential corner cases:
+    # Hebron and Casablanca are chosen because they contain many zone transitions.
+    # Belize is chosen for containing the most intra-hour zone transitions.
+    # Punta_Arenas is chosen for containing the most intra-minute zone transitions.
+    # Morocco is chosen for frequently changing transition rules between the release of Java 8 and Java 17
+    non_utc_time_zones=("Iran" "Hebron" "Casablanca" "Belize" "Punta_Arenas" "Morocco")
+    for tz in "${non_utc_time_zones[@]}"
```

Comment:
Q: do we need to cover all the tzs here in pre-merge? how much more duration it would increase here?

Reply:
Perhaps it will NOT take a long time ? Because we only test 6 test suites for each time zone. Is the `test_parquet_read_round_trip` or `test_read_round_trip` very time-consuming?

Reply:
Yes, but this is likely to change as we add in more support. I would love to see something that is going to be maintainable, but I realize that may need to be a follow on issue.